staging/src/k8s.io/api/scheduling/v1alpha1/types.go

```diff
@@ -72,3 +72,123 @@ type PriorityClassList struct {
 	// items is the list of PriorityClasses
 	Items []PriorityClass `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Workload allows for expressing scheduling constraints that should be used
+// when managing lifecycle of workloads from scheduling perspective,
+// including scheduling, preemption, eviction and other phases.
+type Workload struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	//
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the desired behavior of a Workload.
+	//
+	// +required
+	Spec WorkloadSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// WorkloadList contains a list of Workload resources.
+type WorkloadList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	//
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Items is the list of Workloads.
+	Items []Workload `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// WorkloadSpec defines the desired state of a Workload.
+type WorkloadSpec struct {
+	// ControllerRef is an optional reference to the controlling object, such as a
+	// Deployment or Job. This field is intended for use by tools like CLIs
+	// to provide a link back to the original workload definition.
+	// When set, it cannot be changed.
+	//
+	// +optional
+	ControllerRef *apiv1.TypedLocalObjectReference `json:"controllerRef,omitempty" protobuf:"bytes,1,opt,name=controllerRef"`
+
+	// PodGroups is the list of pod groups that make up the Workload.
+	// This field is immutable.
+	//
+	// +required
+	// +listType=map
+	// +listMapKey=name
```

Comment:
List fields should have an upper-bound -- how many pod groups do you think are needed?